packages/pass-style/src/types.d.ts

```diff
@@ -0,0 +1,182 @@
+/* eslint-disable no-use-before-define */
+import { PASS_STYLE } from './passStyle-helpers.js';
+
+export type PrimitiveStyle =
+  | 'undefined'
+  | 'null'
+  | 'boolean'
+  | 'number'
+  | 'bigint'
+  | 'string'
+  | 'symbol';
+export type ContainerStyle = 'copyRecord' | 'copyArray' | 'tagged';
+export type PassStyle =
+  | PrimitiveStyle
+  | ContainerStyle
+  | 'remotable'
+  | 'error'
+  | 'promise';
+export type PassStyled<S extends string> = {
+  [PASS_STYLE]: S;
+};
+export type ExtractStyle<P extends PassStyled<any>> = P[typeof PASS_STYLE];
+export type PassByCopy =
+  | import('type-fest').Primitive
+  | Error
+  | any[]
+  | CopyRecord
+  | CopyTagged;
+export type PassByRef =
+  | RemotableObject
+  | Promise<RemotableObject>
+  | Promise<PassByCopy>;
+/**
+ * A Passable is acyclic data that can be marshalled. It must be hardened to
+ * remain
+ * stable (even if some components are proxies; see PureData restriction below),
+ * and is classified by PassStyle:
+ *   * Atomic primitive values have a PrimitiveStyle (PassStyle
+ *     'undefined' | 'null' | 'boolean' | 'number' | 'bigint'
+ *     | 'string' | 'symbol').
+ *   * Containers aggregate other Passables into
+ *     * sequences as CopyArrays (PassStyle 'copyArray'), or
+ *     * string-keyed dictionaries as CopyRecords (PassStyle 'copyRecord'), or
+ *     * higher-level types as CopyTaggeds (PassStyle 'tagged').
+ *   * PassableCaps (PassStyle 'remotable' | 'promise') expose local values to
+ *     remote interaction.
+ *   * As a special case to support system observability, error objects are
+ *     Passable (PassStyle 'error').
+ *
+ * A Passable is essentially a pass-by-copy superstructure with a
+ * pass-by-reference
+ * exit point at the site of each PassableCap (which marshalling represents
+ * using 'slots').
+ */
+export type Passable<
+  PC extends PassableCap = PassableCap,
+  E extends Error = Error,
```

Comment:
Took me awhile to understand what's going on here. This is awesome! No change suggested.